### PR TITLE
Feature/kernel compute mean

### PR DESCRIPTION
**Pull request by tc85324**
### PR Type
Closes: #617

- Feature
- Tests

### Description
add `compute_mean` to `coreax.kernel.Kernel`
    
The new `compute_mean` method generalizes the functionality of `gramian_row_mean`, allowing any kernel matrix mean to be computed block-wise. When data is weighted, the mean is computed in the same way as `jax.numpy.average`.

In the future, the `compute_mean` method should be used to remove what is now duplicated functionality from `coreax.metrics.MMD`.

Some additional adjustments were made to `Data` to facilitate these changes. See commit message for further information.

### How Has This Been Tested?
All tests pass locally

### Does this PR introduce a breaking change?
`Data.weights` default to an un-normalized vector of ones with length equal to `Data.data`; prior to this commit, the default weights were normalized.

### Checklist before requesting a review
- [x] I have made sure that my PR is not a duplicate.
- [x] My code follows the style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have performed a self-review of my code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: coreax/kernel.py [hunk 1]
```diff
@@ -229,81 +232,139 @@ def divergence_x_grad_y_elementwise(self, x: ArrayLike, y: ArrayLike) -> Array:
...
-        we accumulate the mean in blocks (memory cost :math:`\mathcal{O}(B^2)`, where
-        ``B`` is a user-specified block size).
+        To avoid materializing the entire matrix (memory cost :math:`\mathcal{O}(n m)),
+        we accumulate the mean over blocks (memory cost :math:`\mathcal{O}(B_x B_y)`,
+        where ``B_x`` and ``B_y`` are user-specified block-sizes for blocking the ``x``
+        and ``y`` parameters respectively.
+
+        .. note:
+            The data ``x`` and/or ``y`` are padded with zero-valued and zero-weighted
+            data points, when ``B_x`` and/or ``B_y``an non-integer divisors of ``n``
```
Comment (db091756) [inline coreax/kernel.py:293]: "are" instead of "an"?

File: tests/unit/test_data.py [hunk 2]
```diff
@@ -68,6 +71,14 @@ def test_len(self, data_type):
         _data = data_type()
         assert len(_data) == len(_data.data)
 
+    @pytest.mark.parametrize("weights", (None, 3, DATA_ARRAY.reshape(-1)))
+    def test_normalize(self, data_type, weights):
```
Comment (db091756) [inline tests/unit/test_data.py:75]: normalise british english?

Comment (tc85324) [reply tests/unit/test_data.py:75]: If you search `normalise` it redirects to  https://www.oed.com/dictionary/normalize_v (with a z)

File: coreax/data.py [hunk 3]
```diff
@@ -201,34 +201,39 @@ class Data(eqx.Module):
...
         return len(self.data)
 
+    def normalize(self) -> Data:
+        """Return a copy of 'self' with 'weights' that sum to one."""
+        normalized_weights = self.weights / jnp.sum(self.weights)
+        return eqx.tree_at(lambda x: x.weights, self, normalized_weights)
+
+
+def is_data(x: Any | Data):
+    """Return 'True' if element is and instance of 'coreax.data.Data'."""
```
[the remark this answers is not in the document]
Comment (tc85324) [reply coreax/data.py:234]: Thanks for catching this!

File: tests/unit/test_data.py (hunk 2 above)
Comment (db091756) [reply tests/unit/test_data.py:75]: What has gone wrong at the OED

**Review by db091756 — DISMISSED** [review]

**Review by db091756 — DISMISSED** [review]

**Review by db091756 — DISMISSED** [review]

**Review by db091756 — APPROVED** [review]
